doc/source/train/examples/pytorch/distributing-pytorch/README.ipynb

```diff
@@ -0,0 +1,599 @@
+    "\n",
+    "                test_loss += loss.item()\n",
+    "                num_total += y.shape[0]\n",
+    "                num_correct += (pred.argmax(1) == y).sum().item()\n",
+    "\n",
+    "        test_loss /= len(train_dataloader)\n",
+    "        accuracy = num_correct / num_total\n",
+    "\n",
+    "        # [3] Report metrics to Ray Train\n",
+    "        # ===============================\n",
+    "        with tempfile.TemporaryDirectory() as temp_checkpoint_dir:\n",
+    "            torch.save(\n",
+    "                model.module.state_dict(),\n",
+    "                os.path.join(temp_checkpoint_dir, \"model.pt\")\n",
+    "            )\n",
+    "            ray.train.report(\n",
+    "                metrics={\"loss\": test_loss, \"accuracy\": accuracy},\n",
+    "                checkpoint=ray.train.Checkpoint.from_directory(temp_checkpoint_dir),\n",
+    "            )\n",
+    "            if ray.train.get_context().get_world_rank() == 0:\n",
+    "                print({\"epoch_num\": epoch, \"loss\": test_loss, \"accuracy\": accuracy})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, run the training function on the Ray Cluster with a TorchTrainer using GPU workers.\n",
+    "\n",
+    "The `TorchTrainer` takes the following arguments:\n",
+    "* `train_loop_per_worker`: the training function you defined earlier. Each Ray Train worker runs this function. Note that you can call special Ray Train functions from within this function.\n",
+    "* `train_loop_config`: a hyperparameter dictionary. Each Ray Train worker calls its `train_loop_per_worker` function with this dictionary.\n",
+    "* `scaling_config`: a configuration object that specifies the number of workers and compute resources (for example, CPUs or GPUs) your training run needs.\n",
+    "* `run_config`: a configuration object that specifies various fields used at runtime, including the path to save checkpoints.\n",
+    "\n",
+    "`trainer.fit` spawns a controller process to orchestrate the training run and worker processes to actually execute the PyTorch training code."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train_cifar_10(num_workers, use_gpu):\n",
+    "    global_batch_size = 512\n",
+    "\n",
+    "    train_config = {\n",
+    "        \"lr\": 1e-3,\n",
+    "        \"epochs\": 1,\n",
+    "        \"batch_size_per_worker\": global_batch_size // num_workers,\n",
+    "    }\n",
+    "\n",
+    "    # [1] Start distributed training\n",
+    "    # Define computation resources for workers\n",
+    "    # Run `train_func_per_worker` on those workers\n",
+    "    # ============================================\n",
+    "    scaling_config = ScalingConfig(num_workers=num_workers, use_gpu=use_gpu)\n",
+    "    run_config = RunConfig(storage_path=\"/mnt/cluster_storage\", name=\"ray_train_run\")\n",
```

Comment:
Create a unique id for each run (ex with a uuid or timestamp), otherwise it will implicitly load some state from the previous run folder.

See this section: https://docs.ray.io/en/latest/train/user-guides/fault-tolerance.html#job-driver-fault-tolerance

Reply:
Do this for the ray data one below too

Reply:
Done.